assembl/static/js/app/views/admin/adminTimelineEvents.js

```diff
@@ -0,0 +1,244 @@
+      titles[loc] = '';
+      descriptions[loc] = '';
+    });
+    title.initFromDict(titles);
+    description.initFromDict(titles);
+    event = new TimelineEvent.Model({
+      title: title,
+      description: description,
+      previous_event: lastEventId,
+    });
+    eventCollection.add(event);
+    event.save();
+    ev.preventDefault();
+  },
+
+  serializeData: function() {
+    return {
+    };
+  },
+
+  onRender: function() {
+    if (this.isViewDestroyed()) {
+      return;
+    }
+    if (this.timelineEventCollection != null) {
+      this.showChildView(
+        "timelineEventsList",
+        new TimelineEventsList({
+          basePanel: this,
+          collection: this.timelineEventCollection,
+        }));
+    }
+    var menu = new AdminNavigationMenu.discussionAdminNavigationMenu(
+      {selectedSection: "timeline"});
+    this.getRegion('navigationMenuHolder').show(menu);
+  },
+});
+
+
+/**
+ * @class  app.views.admin.adminTimelineEvents.TimelineEventView
+ */
+var TimelineEventView = Marionette.LayoutView.extend({
+  constructor: function TimelineEventView() {
+    Marionette.LayoutView.apply(this, arguments);
+  },
+  template: '#tmpl-adminTimelineEvent',
+  ui: {
+    eventTitle: '.js_timeline_title',
+    eventDescription: '.js_timeline_description',
+    eventImageUrl: '.js_timeline_image_url',
+    eventStartDate: '.js_start_date',
+    eventEndDate: '.js_end_date',
+    eventUp: '.js_timeline_up',
+    eventDown: '.js_timeline_down',
+    eventDelete: '.js_timeline_delete',
+  },
+  regions: {
+    eventTitle: '@ui.eventTitle',
+    eventDescription: '@ui.eventDescription',
+  },
+  events: {
+    'click @ui.eventUp': 'reorderColumnUp',
+    'click @ui.eventDown': 'reorderColumnDown',
+    'click @ui.eventDelete': 'deleteColumn',
+    'change @ui.eventImageUrl': 'changeImageUrl',
+    'change @ui.eventStartDate': 'changeStartDate',
+    'change @ui.eventEndDate': 'changeEndDate',
+  },
+  getIndex: function() {
+    return _.indexOf(this.model.collection.models, this.model);
+  },
+  serializeData: function() {
+    return {
+      event: this.model,
+      index: this.getIndex(),
+      collsize: this.model.collection.length,
+    };
+  },
+  onRender: function() {
+    this.showChildView(
+      "eventTitle",
+      new SimpleLangStringEditPanel({
+        model: this.model.get('title'),
+        owner_relative_url: this.model.url() + '/title',
+      }));
+    this.showChildView(
+      "eventDescription",
+      new SimpleLangStringEditPanel({
+        model: this.model.get('description'),
+        owner_relative_url: this.model.url() + '/description',
+      }));
+  },
+  reorderColumnDown: function(ev) {
+    var index = this.getIndex(),
+        nextModel = this.model.collection.at(index + 1);
+    Promise.resolve($.ajax(nextModel.url() + "/reorder_up", {
+        method: "POST"})).then(function(data) {
+        nextModel.collection.fetch({
+          success: function() {
```

Comment:
It's not that important to have a growl, if the AJAX call fails anyway on timeout/server crash. At least there is some indication that the action did not follow through.